File: OxkonWebsites/cestasdelicias.com.br/App_Code/Mail.vb
Imports Microsoft.VisualBasic
Imports System.Net.Mail
Imports System.Net
Imports System.Net.Mime


Public Class Mail

    Function Enviar(ByVal para As String, ByVal cc As String, ByVal de As String, ByVal assunto As String, ByVal msg As String) As String



        Dim objMailMessage As New MailMessage()


        With objMailMessage
            .To.Add(New MailAddress(para))
            .From = New MailAddress(de)
            'Define o assunto 

            .Subject = assunto

            ' Define o corpo da mensagem
            Dim body As String = "<html><head><title>Email</title></head><body><table style=""font-family: Arial, Verdana; font-size: 11px;"" width=""700"" border=""0"" cellpadding=""0"" cellspacing=""0"">  <tr>    <td><img src=""cid:ImagemTop"" /></td>  </tr>  <tr>    <td height=""154"" bgcolor=""#E8E8E8""><blockquote>" + msg + "</blockquote></td>  </tr>  <tr>    <td></p><img src=""cid:ImagemBotton"" /></td>  </tr></table></body></html>"
            Dim view As AlternateView = AlternateView.CreateAlternateViewFromString(body, System.Text.Encoding.UTF8, MediaTypeNames.Text.Html)
            Dim resource As New LinkedResource(System.Configuration.ConfigurationManager.AppSettings("Imagens") + "mailTop.jpg")
            Dim resource2 As New LinkedResource(System.Configuration.ConfigurationManager.AppSettings("Imagens") + "mailBot.jpg")

            resource.ContentId = "ImagemTop"
            resource2.ContentId = "ImagemBotton"
            view.LinkedResources.Add(resource)
            view.LinkedResources.Add(resource2)
            .AlternateViews.Add(view)


            '.Body = body
            ' Define o formato do email como HTML

            .IsBodyHtml = True

            ' Define a prioridade da mensagem como normal

            .Priority = MailPriority.Normal

            If Not cc Is Nothing And cc <> String.Empty Then

                ' Define o endereço cc 

                .CC.Add(New MailAddress(cc))

            End If




        End With

        Try
            Dim mSmtpClient As New SmtpClient("localhost")
            mSmtpClient.UseDefaultCredentials = True
            mSmtpClient.Credentials = New NetworkCredential("zou@zoudesign.com.br", "z1o2u3")
            mSmtpClient.Send(objMailMessage)
            Return ""
        Catch ex As Exception

            Return (ex.Message)
        End Try



    End Function


End Class

File: OxkonWebsites/coletadeentulhovrs.com.br/App_Code/Mail.cs
using System;
using System.Collections.Generic;
using System.Web;
using System.Net.Mail;

/// <summary>
/// Classe para envio de e-mail via SMTP local
/// Deve configurar o Dominio no proprio arquivo Mail.cs
/// Caso o e-mail seja enviado retorna um string vazio
/// </summary>
public class Mail
{
    String Dominio = "@coletadeentulhovrs.com.br";
    public string Enviar(string para, string cc, string de, string assunto, string msg, string Anexo, Boolean UsaLayout)
    {
        MailMessage objMailMessage = new MailMessage();


        //With objMailMessage
        objMailMessage.To.Add(new MailAddress(para));
        objMailMessage.From = new MailAddress(de);

        objMailMessage.Subject = assunto;


        string body = "";
        if (UsaLayout)
        {
            body = "<table width=\"700\" border=\"0\" cellpadding=\"0\" cellspacing=\"0\">  <tr>    <td><img src=\"cid:ImagemTop\" /></td>  </tr>  <tr>    <td height=\"154\" bgcolor=\"#FFFFFF\"><blockquote>" + msg + "</blockquote></td>  </tr>  <tr>    <td></p><img src=\"cid:ImagemBotton\" /></td>  </tr></table>";
            AlternateView view = AlternateView.CreateAlternateViewFromString(body, System.Text.Encoding.UTF8, System.Net.Mime.MediaTypeNames.Text.Html);
            LinkedResource resource = new LinkedResource(System.Configuration.ConfigurationManager.AppSettings["Imagens"] + @"\mail.jpg");
            LinkedResource resource2 = new LinkedResource(System.Configuration.ConfigurationManager.AppSettings["Imagens"] + @"\mailrodape.jpg");
            resource.ContentId = "ImagemTop";
            resource2.ContentId = "ImagemBotton";
            view.LinkedResources.Add(resource);
            view.LinkedResources.Add(resource2);
            objMailMessage.AlternateViews.Add(view);
        }
        else
        {
            objMailMessage.Body = msg;
        }

        objMailMessage.SubjectEncoding = System.Text.Encoding.GetEncoding("utf-8");
        objMailMessage.BodyEncoding = System.Text.Encoding.GetEncoding("utf-8");
        objMailMessage.Headers.Add("Message-ID", "<" + System.Guid.NewGuid().ToString().Replace("-", "") + Dominio + ">");
        objMailMessage.IsBodyHtml = true;

        if (Anexo != "")
            objMailMessage.Attachments.Add(new Attachment(Anexo));

        objMailMessage.Priority = MailPriority.Normal;

        if (!string.IsNullOrEmpty(cc))
        {
            String[] arrayCC = cc.Split(";".ToCharArray());
            foreach (string s in arrayCC)
                objMailMessage.CC.Add(new MailAddress(s));
        }


        try
        {
            SmtpClient mSmtpClient = new SmtpClient("localhost");
            mSmtpClient.Send(objMailMessage);
            return "";
        }
        catch (Exception ex)
        {
            return ex.Message;
        }

    }
}